### ISSUE TITLE
MvxRecyclerView: Exception masked during Adapter RealNotifyDataSetChanged ArgumentException: 'jobject' must not be IntPtr.Zero.

### ISSUE THREAD

**Issue by SlyNet** [MvvmCross/MvvmCross-AndroidSupport#175]
All is needed is to swap the default `MvxRecyclerView` adapter with custom implementation and then change contents of binded `ObservableCollection`. There is no threading or anything other involved. 
Try tapping on _Hello World, Click Me!_ button multimple times and warnings are visible in output window:

```
mvx:Warning: 10.80 Exception masked during Adapter RealNotifyDataSetChanged ArgumentException: 'jobject' must not be IntPtr.Zero.
Parameter name: jobject
      at Android.Runtime.JNIEnv.CallVoidMethod (IntPtr jobject, IntPtr jmethod, Android.Runtime.JValue* parms) [0x00010] in /Users/builder/data/lanes/2512/d3008455/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.g.cs:563 
  at Android.Support.V7.Widget.RecyclerView+Adapter.NotifyItemRangeInserted (Int32 positionStart, Int32 itemCount) [0x0005d] in <filename unknown>:0 
  at MvvmCross.Droid.Support.V7.RecyclerView.MvxRecyclerAdapter.NotifyDataSetChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00038] in <filename unknown>:0 
[0:] mvx:Warning: 10.80 Exception masked during Adapter RealNotifyDataSetChanged ArgumentException: 'jobject' must not be IntPtr.Zero.
Parameter name: jobject
      at Android.Runtime.JNIEnv.CallVoidMethod (IntPtr jobject, IntPtr jmethod, Android.Runtime.JValue* parms) [0x00010] in /Users/builder/data/lanes/2512/d3008455/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.g.cs:563 
  at Android.Support.V7.Widget.RecyclerView+Adapter.NotifyItemRangeInserted (Int32 positionStart, Int32 itemCount) [0x0005d] in <filename unknown>:0 
```

Sample project is attached. 
[MvxRecyclerView1.zip](https://github.com/MvvmCross/MvvmCross-AndroidSupport/files/106282/MvxRecyclerView1.zip)

**Comment by martijn00** [comment]
Isn't that fixed in #167 ?

**Comment by SlyNet** [comment]
Yes, it does. Thanks! Used beta-8 in project, so this patch is not included yet.
